### PR TITLE
Update the app for the visierConfig visierUrl update, the new debug events, and multiple tab error handling

### PR DESCRIPTION
**Pull request by ttingt**
The Embedding Docs are already updated for these changes. Updating this Embedding Exemplar app to match as well.

Changes include:
* In the `visierConfig`, `loginUrl` is now deprecated and replaced with `visierUrl`. The login popup was a legacy workflow that is no longer used so we're cleaning up to simplify confusions around having a `loginUrl`.
* Adding optional logging for the new `debug` events. Set `enableDebugLogging` to turn on console logging for these events to aid with troubleshooting.
* Add error handling for multiple tab or window scenarios. There are 2 new events: 
  * `EJECT_SESSION` when a previous session is ejected/cleaned up to make way for the session of a different tenant/user. Example recording below.  
  * `FAILED_TO_EJECT_PREVIOUS_SESSION` when the current tab or window is setting up a different tenant/user but we fail to fully clean up the previous session on a prior tab or window.

https://github.com/visier/embedding-exemplar/assets/8945152/fe6da094-5ef3-4f46-9573-86087892feea

### CODE REVIEW

**Review by dontsteponme — APPROVED** [review]
Looks good. Thank you

File: public/embed-application.js [hunk 1]
```diff
@@ -128,7 +135,21 @@ function cleanUpVisierSession() {
  */
 function attachSessionEventHandlers(embeddingApp) {
     embeddingApp.on("session", function(msg) {
+        console.log("SESSION: ", msg);
         switch (msg?.code?.toUpperCase()) {
+            case "EJECT_SESSION":
+                /**
+                 * This message is emitted if another tab is opened which embeds content for a different user or tenant.
```
Comment (emily-watt-visier) [inline public/embed-application.js:142]: This message is emitted if the user opens another tab that embeds content for a different user or tenant.

File: public/embed-application.js [hunk 2]
```diff
@@ -199,11 +223,22 @@ function attachSessionEventHandlers(embeddingApp) {
...
     embeddingApp.on("error", function(msg) {
+        console.error("ERROR: ", msg);
         switch (msg?.code?.toUpperCase()) {
             case "CONTAINER_MISSING":
                 // If the container to embed the content in cannot be found.
                 renderVisierErrorMessage("Invalid container", msg);
                 break;
+            case "FAILED_TO_EJECT_PREVIOUS_SESSION":
+                /**
+                 * When this is a second page that was opened that embeds content for a different user or tenant,
```
Comment (emily-watt-visier) [inline public/embed-application.js:234]: If the user opens a second tab that embeds content for a different user or tenant, this error is emitted in the case of unexpected errors while performing cleanup on the first tab.

File: public/embed-chart.html [hunk 3]
```diff
@@ -144,6 +166,18 @@ <h3 class="subtitle">The partner application embeds selected Visier visuals into
         function attachSessionEventHandlers(embeddingApp) {
             embeddingApp.on("session", function(msg) {
                 switch (msg?.code?.toUpperCase()) {
+                    case "EJECT_SESSION":
+                        /**
+                         * This message is emitted if another tab is opened which embeds content for a different user or tenant.
```
Comment (emily-watt-visier) [inline public/embed-chart.html:171]: Same change as above

File: public/embed-chart.html [hunk 4]
```diff
@@ -199,6 +233,27 @@ <h3 class="subtitle">The partner application embeds selected Visier visuals into
             embeddingApp.on("error", function(msg) {
                 // Refer to the following list for common error cases.
                 switch (msg?.code) {
+                    /**
+                     * When this is a second page that was opened that embeds content for a different user or tenant,
```
Comment (emily-watt-visier) [inline public/embed-chart.html:237]: Same change as above

File: public/embed-chart.html [hunk 5]
```diff
@@ -199,6 +233,27 @@ <h3 class="subtitle">The partner application embeds selected Visier visuals into
...
+                     * The partner application can clean up the other session and attempt embedding once again.
+                     */
+                    case "FAILED_TO_EJECT_PREVIOUS_SESSION":
+                        showErrorModal("The user is logged into another tenant on another tab or window.");
+                        hideCharts();
+                        break;
+                    /**
+                     * This timeout error is emitted when there are unexpected issues with the authentication set up or if
+                     * the set up took longer than expected such as due to network latency. Possible causes include:
+                     *  - The IdP failed to authenticate the user. Check the network tab.
```
Comment (emily-watt-visier) [inline public/embed-chart.html:250]: Check the Network tab.

File: public/embed-chart.html [hunk 6]
```diff
@@ -199,6 +233,27 @@ <h3 class="subtitle">The partner application embeds selected Visier visuals into
...
+                     * errors performing that clean up on the other page.
+                     *
+                     * The partner application can clean up the other session and attempt embedding once again.
+                     */
+                    case "FAILED_TO_EJECT_PREVIOUS_SESSION":
+                        showErrorModal("The user is logged into another tenant on another tab or window.");
+                        hideCharts();
+                        break;
+                    /**
+                     * This timeout error is emitted when there are unexpected issues with the authentication set up or if
```
Comment (emily-watt-visier) [inline public/embed-chart.html:248]: This timeout error emits if there are unexpected issues with the authentication set up or if the set up took longer than expected, such as due to network latency. Possible causes include:

File: public/embed-chart.html [hunk 7]
```diff
@@ -199,6 +233,27 @@ <h3 class="subtitle">The partner application embeds selected Visier visuals into
...
+                     * the set up took longer than expected such as due to network latency. Possible causes include:
+                     *  - The IdP failed to authenticate the user. Check the network tab.
+                     *  - The service worker failed to load. Look for errors in the console and turn on enableDebugLogging.
```
Comment (emily-watt-visier) [inline public/embed-chart.html:251]: if this is the Console tab, let's say: The service worker failed to load. Look for errors in the Console tab and turn on enableDebugLogging.

**Review by emily-watt-visier — APPROVED** [review]
Added a few suggestions that you can take or leave

**Pull request merged**